azure-devops/templates/test_steps.yml

```diff
@@ -0,0 +1,249 @@
+
+steps:
+  # we are using prebuild artifacts, so we do not need to checkout
+  - checkout: none
+
+  - script: |
+      git clone --branch $ADOPT_JDK_SOURCE_BRANCH --depth 1 $ADOPT_JDK_SOURCE_URI $(Build.SourcesDirectory)
+    displayName: 'Clone jdk sources'
+
+  - bash: |
+      ERR_MESSAGE_MISSING_ENV=''
+
+      if [ -z "$DEBUG_LEVEL"  ]; then
+        ERR_MESSAGE_MISSING_ENV="Pipeline must set DEBUG_LEVEL to fastdebug, slowdebug, release. $ERR_MESSAGE_MISSING_ENV"
+      fi;
+
+      if [ -z "$BUILD_LIST"  ]; then
+        ERR_MESSAGE_MISSING_ENV="Pipeline must set BUILD_LIST to [openjdk|system|external|etc...]. $ERR_MESSAGE_MISSING_ENV"
+      fi;
+
+      if [ -z "$TEST_TARGET"  ]; then
+        ERR_MESSAGE_MISSING_ENV="Pipeline must set TEST_TARGET to _[sanity|extended|special|etc.].suite_name $ERR_MESSAGE_MISSING_ENV"
+      fi;
+
+      if [ -z "$ADOPT_JDK_SOURCE_URI"  ]; then
+        ERR_MESSAGE_MISSING_ENV="Pipeline must set ADOPT_JDK_SOURCE_URI to the uri of the jdk repo to use. $ERR_MESSAGE_MISSING_ENV"
+      fi;
+
+      if [ -z "$ADOPT_JDK_SOURCE_BRANCH"  ]; then
+        ERR_MESSAGE_MISSING_ENV="Pipeline must set ADOPT_JDK_SOURCE_BRANCH to the branch-name of the jdk repo to use. $ERR_MESSAGE_MISSING_ENV"
+      fi;
+
+      if [ -z "$ADOPT_TEST_REPO_URI"  ]; then
+        ERR_MESSAGE_MISSING_ENV="Pipeline must set ADOPT_TEST_REPO_URI to the uri of the openjdk-tests repo to use. $ERR_MESSAGE_MISSING_ENV"
+      fi;
+
+      if [ -z "$ADOPT_TEST_REPO_BRANCH"  ]; then
+        ERR_MESSAGE_MISSING_ENV="Pipeline must set ADOPT_TEST_REPO_BRANCH to the branch-name of the openjdk-tests repo to use. $ERR_MESSAGE_MISSING_ENV"
+      fi;
+
+      if [ -z "$TEST_NAME"  ]; then
+        ERR_MESSAGE_MISSING_ENV="Pipeline must set TEST_NAME to the tests to run. $ERR_MESSAGE_MISSING_ENV"
+      fi;
+
+      if [ -z "$FAIL_TASK_ON_FAILED_TESTS"  ]; then
+        ERR_MESSAGE_MISSING_ENV="Pipeline must set FAIL_TASK_ON_FAILED_TESTS to true or false. $ERR_MESSAGE_MISSING_ENV"
+      fi;
+
+      if [ -z "$TEST_RESULT_SEARCH_PATH"  ]; then
+        ERR_MESSAGE_MISSING_ENV="Pipeline must set TEST_RESULT_SEARCH_PATH to the local search path for test report files. $ERR_MESSAGE_MISSING_ENV"
+      fi;
+
+      if [ ! -z "$ERR_MESSAGE_MISSING_ENV" ]; then
+        echo "##vso[task.logissue type=error;]$ERR_MESSAGE_MISSING_ENV"
+        echo "##vso[task.complete result=Failed;]"
+      fi;
+
+    displayName: Check for required parameters
+
+  - script: |
+      brew update
+    displayName: 'Update brew'
+    condition: and(succeeded(), eq(variables['Agent.OS'], 'Darwin'))
+
+  # Install macOS required dependencies
+  - script: |
+      set -e
+      brew install xmlstarlet ant ant-contrib tree gnu-sed
+    displayName: "macOS: install dependencies"
+    condition: and(succeeded(), eq(variables['Agent.OS'], 'Darwin'))
+
+  - script: |
+      export PERL_MM_USE_DEFAULT=1
+      cpan install JSON Text::CSV XML::Parser
```

Comment:
No longer need the special Perl modules (since this PR was merged which among other things removes that dependency: https://github.com/eclipse/openj9/pull/7398

Based on our Slack conversation last week, I have some other ideas on this work, which I can start pulling together for sharing this week.  :)

Reply:
removed in https://github.com/AdoptOpenJDK/openjdk-build/pull/1306/commits/026de7d387d1031043875550fee13616e0cd6adb